### PR TITLE
uninstall: add switching to the latest installed version

### LINKED ISSUE #514: uninstall: can't uninstall version recently installed
Consider the following scenario:
```
$ ../../Projects/tt/tt install tarantool 2.10.7
...
   • Done.  
$ ../../Projects/tt/tt install tarantool 2.10.8-entrypoint
...
   • Done. 
$ ../../Projects/tt/tt uninstall tarantool 2.10.8-entrypoint
...
   • tarantool=2.10.8-entrypoint is uninstalled.
$ ../../Projects/tt/tt uninstall tarantool 2.10.7
   • Removing binary...       
   ⨯ failed to resolve symlink /home/askalt/Samples/tt/bin/tarantool: lstat /home/askalt/Samples/tt/bin/tarantool: no such file or directory
```

Installed tarantool 2.10.7 cannot be removed.

### PR DESCRIPTION
**Pull request by askalt**
If there are other versions of the program present when removing the symlink, the symlink should be switched to one of them; otherwise, `bin_dir` and `include_dir` end up in an incorrect state. Added switching to the latest installed version.

Closes #514

### CODE REVIEW

**Comment by askalt** [comment]
```
$ tt uninstall tarantool 1.10.15
   • Removing binary...       
   • Removing headers...      
   • tarantool=1.10.15 is uninstalled.
   • Changing symlinks...     
   • tarantool binary was set in tarantool_2.10.4.

$ tree bin/ include/
bin/
├── tarantool -> tarantool_2.10.4
└── tarantool_2.10.4
include/
└── include
    ├── tarantool -> tarantool_2.10.4
    └── tarantool_2.10.4

3 directories, 0 files

```

File: cli/uninstall/uninstall.go [hunk 1]
```diff
@@ -26,53 +26,56 @@ const (
...
 	}
 	// Get path where symlink point.
 	resolvedPath, err := util.ResolveSymlink(linkPath)
 	if err != nil {
-		return fmt.Errorf("failed to resolve symlink %s: %s", linkPath, err)
+		return false, fmt.Errorf("failed to resolve symlink %s: %s", linkPath, err)
 	}
+	var isSymlinkRemoved bool
 	// Remove symlink if it points to program.
 	if strings.Contains(resolvedPath, fileName) {
```
Comment (better0fdead) [inline cli/uninstall/uninstall.go:66]: i think it is better to use strings.HasSuffix()

File: cli/uninstall/uninstall.go [hunk 2]
```diff
@@ -26,53 +26,56 @@ const (
...
 		if err != nil {
-			return err
+			return false, err
 		}
+		isSymlinkRemoved = true
 	}
 	err = os.RemoveAll(path)
 	if err != nil {
-		return err
+		return isSymlinkRemoved, err
```
Comment (better0fdead) [inline cli/uninstall/uninstall.go:75]: What will happen if symlink is removed but os.RemoveAll(path) will fail? Symlink will be removed but program won't.
Should we handle this case somewhat different?

File: cli/uninstall/uninstall.go [hunk 3]
```diff
@@ -173,3 +186,93 @@ func GetList(cliOpts *config.CliOpts) []string {
...
+
+	binaries, err := os.ReadDir(binDst)
+	if err != nil {
+		return "", err
+	}
+
+	latestVersionInfo := version.Version{}
+	latestVersion := ""
+
+	for _, binary := range binaries {
```
Comment (better0fdead) [inline cli/uninstall/uninstall.go:211]: Do we check here if entry is not a directory?

File: test/integration/uninstall/testdata/uninstall_switch/tt.yaml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+tt:
+  app:
+    bin_dir: "./bin"
+    inc_dir: "./inc"
```
Comment (better0fdead) [inline test/integration/uninstall/testdata/uninstall_switch/tt.yaml:4]: no empty line at the end of file

**Review by better0fdead — CHANGES_REQUESTED** [review]
Hi. Thanks for the patch. See small comments bellow.

File: cli/uninstall/uninstall.go (hunk 1 above)
Comment (askalt) [reply cli/uninstall/uninstall.go:66]: This change will not be directly related to this task, so I think it's better to keep it unchanged.

File: cli/uninstall/uninstall.go (hunk 2 above)
Comment (askalt) [reply cli/uninstall/uninstall.go:75]: In this case, uninstall switching won't happen, and `tt` will print error, i.e., the behavior will not change.

File: cli/uninstall/uninstall.go (hunk 3 above)
Comment (askalt) [reply cli/uninstall/uninstall.go:211]: Fixed. Now check it inside cycle.

File: test/integration/uninstall/testdata/uninstall_switch/tt.yaml (hunk 4 above)
Comment (askalt) [reply test/integration/uninstall/testdata/uninstall_switch/tt.yaml]: Fixed.

**Review by better0fdead — APPROVED** [review]

File: cli/uninstall/uninstall_test.go [hunk 5]
```diff
@@ -52,3 +52,60 @@ func TestGetList(t *testing.T) {
...
+			binDst:      "./testdata/bin_basic",
+			headerDst:   "./testdata/inc_invalid",
+			expectedVer: "tt_2.0.0",
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			ver, err := searchLatestVersion(tc.linkName, tc.binDst, tc.headerDst)
+			assert.NoError(t, err)
```
Comment (psergee) [inline cli/uninstall/uninstall_test.go:107]: Check error case: use file name as a bin dir, `readdir` will fail.

File: cli/uninstall/uninstall.go [hunk 6]
```diff
@@ -173,3 +186,96 @@ func GetList(cliOpts *config.CliOpts, program string) []string {
...
+
+	if linkName == "tarantool" {
+		headerPath := filepath.Join(headerDst, linkName)
+		err = util.CreateSymlink(filepath.Join(headerDst, progToSwitch), headerPath, true)
+		if err != nil {
+			return err
+		}
+	}
+
+	log.Infof("%s binary was set in %s.", linkName, progToSwitch)
```
Comment (psergee) [inline cli/uninstall/uninstall.go:279]:
```suggestion
	log.Infof("Current %q is set to %q.", linkName, progToSwitch)
```

**Review by psergee — APPROVED** [review]
Thanks for the patch! lgtm.

**Pull request merged**